### DISCUSSION TITLE
[Script request] HA Assist tools

### DISCUSSION THREAD

**Discussion by titiviking** [community-scripts/ProxmoxVE#882] [Request script]
### Application Name

Wyoming Piper/Whisper/OpenWakeWord

### Website

https://www.home-assistant.io/integrations/wyoming/

### Description

Script request for LXC deployment that includes HA Assist tools **with GPU support** to accelerate processing and improve response times.
Tools in scope:
- Piper
- OpenWakeWord
- Whisper

All tools should support Wyoming protocol to ensure native integration in HA.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by spijkey87** [comment]
With gpu!!!!!!

**Comment by jsapede** [comment]
With openvino (Whisper.cpp)

**Comment by MickLesk** [comment]
Not sure what to do here? Is there a source code? Is there a project for this? Where can we crawl data? Where can we synchronize RELEASES? Just a link from the HA docu does nothing for me.

In addition, it is not clear to me what exactly this is supposed to do as LXC, since everything is available as an add-on and integration in HA. Why should you put in development and maintenance effort (for the next months / years)?

**Comment by titiviking** [comment]
Currently there are only 2 deployment options for the mentioned software:
- native python installation
- docker images

The python installations offer the most flexibility, but are complet to deploy and the combination with Wyoming is not always present.
The docker images are very convenient, but don't have GPU support.
Images can be rebuild, but that makes it harder to maintain.

The request consist of having the native python installations on an LXC. It avoid the need of maintaining images and avoids yet another layer of virtualization/containerization.
Also I"m not experience into configuring Python to leverage GPU... hence the LXC script approach looks very appealing.

Ideally a script should be released that would deploy an LXC with configured python environments and the relevant libraries for Piper, Whisper & OpenWakeWord. All supporting GPU acceleration and Wyoming protocol for easy integration into Home Assistant once the LXC is deployed.

Hope this helps to clarify the request.
